src/ZODB/Connection.py

```diff
@@ -101,11 +103,10 @@ def __init__(self, db, cache_size=400, before=None, cache_size_bytes=0):
         # Multi-database support
         self.connections = {self._db.database_name: self}
 
-        storage = db._mvcc_storage
         if before:
-            storage = storage.before_instance(before)
+            storage = HistoricalStorageAdapter(db.storage, before)
```

Comment:
OK, I came up with something that seems pretty sane, and allows me to keep loadBefore out of MVCCAdapterInstance. :)